homeassistant/components/goalzero/binary_sensor.py

```diff
@@ -60,8 +55,18 @@ def unique_id(self):
         """Return the unique id of the sensor."""
         return f"{self._server_unique_id}/{self._condition_name}"
 
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return self._icon
+
```

Comment:
Drop the `icon` as it will get set by the `device_class`

Reply:
mdi:clock-digital will be used for the backlight. Wouldn't I need this or something similar?

Reply:
If there is no device_class in this case, then it should be fine to keep 👍 

If there is a device_class, it should be set to `None`

Reply:
The 3 port status entities are actually switches but I'm currently leaving them as binary sensors. These switches are controllable with a post request. However, posting a command activates the wifi hotspot used for setup. After a while, the device then disconnects from the network. We obviously don't want that. FYI